iceoryx_posh/test/moduletests/test_roudi_portpool.cpp

```diff
@@ -0,0 +1,604 @@
+
+    ASSERT_EQ(applicationPortDataList.size(), 0U);
+}
+
+TEST_F(PortPool_test, AddConditionVariableDataIsSuccessful)
+{
+    auto conditionVariableData = sut.addConditionVariableData(m_applicationName);
+
+    ASSERT_EQ(conditionVariableData.value()->m_process, m_applicationName);
+}
+
+TEST_F(PortPool_test, AddConditionVariableDataWithMaxCapacityIsSuccessful)
+{
+    cxx::vector<popo::ConditionVariableData*, MAX_NUMBER_OF_CONDITION_VARIABLES> conditionVariableContainer;
+
+    for (uint32_t i = 1U; i <= MAX_NUMBER_OF_CONDITION_VARIABLES; ++i)
+    {
+        auto conditionVariableData = sut.addConditionVariableData(m_applicationName);
+        if (!conditionVariableData.has_error())
+        {
+            conditionVariableContainer.push_back(conditionVariableData.value());
+        }
+    }
+
+    ASSERT_EQ(conditionVariableContainer.size(), MAX_NUMBER_OF_CONDITION_VARIABLES);
+}
+
+TEST_F(PortPool_test, AddConditionVariableDataWhenContainerIsFullReturnsError)
+{
+    auto errorHandlerCalled{false};
+    Error errorHandlerType;
+    auto errorHandlerGuard =
+        ErrorHandler::SetTemporaryErrorHandler([&](const Error error, const std::function<void()>, const ErrorLevel) {
+            errorHandlerType = error;
+            errorHandlerCalled = true;
+        });
+
+    for (uint32_t i = 0U; i <= MAX_NUMBER_OF_CONDITION_VARIABLES; ++i)
+    {
+        auto conditionVariableData = sut.addConditionVariableData(m_applicationName);
+    }
+
+    ASSERT_TRUE(errorHandlerCalled);
+    ASSERT_EQ(errorHandlerType, Error::kPORT_POOL__CONDITION_VARIABLE_LIST_OVERFLOW);
+}
+
+TEST_F(PortPool_test, GetConditionVariableDataListIsSuccessful)
+{
+    sut.addConditionVariableData(m_applicationName);
+    auto condtionalVariableData = sut.getConditionVariableDataList();
+
+    ASSERT_EQ(condtionalVariableData.size(), 1U);
+}
+
+TEST_F(PortPool_test, GetConditionVariableDataListWhenEmptyIsSuccessful)
+{
+    auto condtionalVariableData = sut.getConditionVariableDataList();
+
+    ASSERT_EQ(condtionalVariableData.size(), 0U);
+}
+
+TEST_F(PortPool_test, GetConditionVariableDataListCompletelyFilledIsSuccessful)
+{
+    for (uint32_t i = 0; i < MAX_NUMBER_OF_CONDITION_VARIABLES; ++i)
+    {
+        ProcessName_t applicationName = {cxx::TruncateToCapacity, "AppName" + std::to_string(i)};
+        sut.addConditionVariableData(applicationName);
+    }
+    auto condtionalVariableData = sut.getConditionVariableDataList();
+
+    ASSERT_EQ(condtionalVariableData.size(), MAX_NUMBER_OF_CONDITION_VARIABLES);
+}
+
+TEST_F(PortPool_test, RemoveConditionVariableDataIsSuccessful)
+{
+    auto conditionVariableData = sut.addConditionVariableData(m_applicationName);
+
+    sut.removeConditionVariableData(conditionVariableData.value());
+    auto condtionalVariableData = sut.getConditionVariableDataList();
+
+    ASSERT_EQ(condtionalVariableData.size(), 0U);
+}
+
+TEST_F(PortPool_test, GetServiceRegistryChangeCounterIsSuccessful)
```

Comment:
This tests against the initial value of the `serviceRegistryChangeCounter` so the name needs to be adapted, would be good to have a test that return something else than null.

Reply:
done, changed name and added new test